recipes/recipe-5-seeking-to-arbitrary-offsets-kafka.adoc

```diff
@@ -0,0 +1,98 @@
+# Seeking to arbitrary offsets in Kafka
+
+## Problem Statement
+
+Using Kafka binder, I know that it can set the offset to either `earliest` or `latest`, but I have a requirement to seek the offset to something in the middle, an arbitrary offset.
+Is there a way to achieve this using Spring Cloud Stream Kafka biner?
+
+## Solution
+
+Previously we saw how Kafka binder allows you to tackle basic offset management.
+By default, the binder does not allow you to rewind to an arbitrary offset, at least through the mechanism we saw in that reipce.
+However, there are some low-level strategies that the binder provides to achieve this use case.
+Let's explore them.
+
+First of all, when you want to reset to an arbitrary offset other than `earliest` or `latest`, make sure to leave the `resetOffsets` configuration to its defaults, which is `false`.
+Then you have to provide a custom bean of type `KafkaBindingRebalanceListener`, which will be injected into all consumer bindings.
+It is an interface that comes with a few default methods, but here is the method that we are interested in:
+
+```
+/**
+	 * Invoked when partitions are initially assigned or after a rebalance. Applications
+	 * might only want to perform seek operations on an initial assignment. While the
+	 * 'initial' argument is true for each thread (when concurrency is greater than 1),
+	 * implementations should keep track of exactly which partitions have been sought.
+	 * There is a race in that a rebalance could occur during startup and so a topic/
+	 * partition that has been sought on one thread may be re-assigned to another
+	 * thread and you may not wish to re-seek it at that time.
+	 * @param bindingName the name of the binding.
+	 * @param consumer the consumer.
+	 * @param partitions the partitions.
+	 * @param initial true if this is the initial assignment on the current thread.
+	 */
+	default void onPartitionsAssigned(String bindingName, Consumer<?, ?> consumer,
+			Collection<TopicPartition> partitions, boolean initial) {
+		// do nothing
+	}
+```
+
+Let us look at the details.
```

Comment:
Not sure, but it appears that something is up with rendering, like if there are missing formatting somewhere